src/client/client_test.go

```diff
@@ -210,3 +211,26 @@ func Test_createRequest_Error(t *testing.T) {
 	_, err := createRequest(endpoint)
 	assert.Error(t, err)
 }
+
+func TestClientTimeout(t *testing.T) {
+	defaultClient = nil
+	args.GlobalArgs = args.RabbitMQArguments{
+		Timeout: 2,
+	}
+	mux, teardown := testutils.GetTestServer(false)
+	defer teardown()
+	mux.HandleFunc(ConnectionsEndpoint, func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(5 * time.Second)
+	})
+
+	err := CollectEndpoint("", nil)
+	assert.Error(t, err)
+
+	err = CollectEndpoint(ConnectionsEndpoint, nil)
+	assert.Error(t, err)
```

Comment:
Nit: as this is the client-timeout test case, would it make sense to move those error checks to other test function? (or just simplify this test case by removing those lines here)

Reply:
Good point. Solved in [#6011e49](https://github.com/newrelic/nri-rabbitmq/pull/99/commits/6011e49e94aec9e397a767e6ba8b847a149ca726)